### ISSUE TITLE
IDE0251 incorrectly reported when using primary constructors

### ISSUE THREAD

**Issue by jbevain** [dotnet/roslyn#70780]
**Version Used**: 

VS Code 1.84
C# v2.10.28
C# Dev Kit v1.1.9
.NET 8.0 RC 2

**Steps to Reproduce**:

1. Create a new console projet, write the following:

```csharp
Console.WriteLine("Hello, World!");

struct Cell(short value)
{
    public readonly short Value => value;

    public void RemoveBit(int candidate)
    {
        value = (short)(value & ~(1 << candidate));
    }
}
```

**Diagnostic Id**: IDE0251

**Expected Behavior**:

The method RemoveBit modifies the struct and should not be suggested to be made readonly.

**Actual Behavior**:

The method RemoveBit is suggested to be made readonly.
